### DISCUSSION TITLE
How can I programmatically determine if the OpenTelemetry SDK is initialized/enabled?

### DISCUSSION THREAD

**Discussion by w1am** [open-telemetry/opentelemetry-java#6828] [Q&A]
I am manually instrumenting my library by injecting the trace ID and span ID directly into the code. These values are then used later for various operations. However, I want to avoid injecting this trace context if OpenTelemetry is not initialized or enabled when users are using the library.

Is there an easy way to achieve this behavior? The only solution I can think of is to create a separate instrumentation package that users can install, instead of handling the instrumentation directly within the library.

**Comment by trask** [comment]
hi @w1am! if you allow users to provide their own `OpenTelemetry` instance, and fallback to `OpenTelemetry.noop()` when they don't provide their own instance, does this solve the issue for you?
